integration_tests/array_01_decl.py

```diff
@@ -9,6 +9,10 @@ def accept_i64_array(xi64: i64[:]) -> i64:
     xi64[0] = 64
     return xi64[0]
 
+def accept_int_array(xint: int[:]) -> i64:
+    xint[0] = 64
+    return xint[0]
+
```

Comment:
This works with LPython but not with CPython. CPython throws the following error,

```bash
Traceback (most recent call last):
  File "/Users/czgdp1807/lpython_project/lpython/integration_tests/array_01_decl.py", line 12, in <module>
    def accept_int_array(xint: int[:]) -> i64:
TypeError: 'type' object is not subscriptable
```

So I pushed https://github.com/lcompilers/lpython/pull/1086/commits/8336f96d0dec306687cb66b4a0d95444c813d261. Using `py_float`, `py_int` seems a bit hackish but then the alternative is to not support `x: float[:]`, `x: int[:]`, which makes reduces the significance of `x: float` use case. Like if users can only declare normal variables and not arrays using floating point annotations then its not worth it to support `float` annotation. What do you say @certik ?

Reply:
Is there any other way to achieve this (i.e., make the `float[:]` work with CPython)?

Reply:
Why not something like `Array[int]`? `int[:]` might not be in the spirit of PEP 483.

I general, we should *never* have something that works in LPython and not in CPython. That's an extension of Python and is a road to perdition! The reverse is ok IMO.

Reply:
`Array[int]` makes sense. For multi-dimensional arrays, should it look like, `Array[Array[int]]`?

Reply:
This is a more complex issue, so I created https://github.com/lcompilers/lpython/issues/1107 where we can discuss it more. In short: `Array[int, :, :]` can probably be done, but it's quite verbose, so we should first decide if we want to go this route, in addition (or even instead!) of the short `i64[:,:]` syntax. Also note that `int` in CPython means arbitrary precision integer, so we should not use it interchangeably with `i64`. However, `float` in CPython means `f64`, so that could be done, for scalars. For arrays of `f64`, the same issue happens. See #1107 for more discussion.